### DISCUSSION TITLE
Aktivierung openDTU abhängig von Startreihenfolge fehlerhaft

### DISCUSSION THREAD

**Discussion by Fokker-1** [tbnobody/OpenDTU#1168] [Q&A]
Hallo,
dies ist mein erster Post hier.
Zuvor aber meinen ganz großen Dank an die klugen und engagierten Köpfe hier für das openDTU-Projekt - super!

Seit knapp einem Monat läuft meine PV-Anlage mit einem HM-800 absolut problemlos - bis auf den folgenden Punkt und b).

Anordnung:
- Der HM-800 ist per WLAN mit einer FritzBox 6591cable verbunden.
- In der FB ist die WLAN-Zeitschaltung zur Stromsparnis über Nacht aktiviert.
- Die Betriebsspannung der DTU ist auf der 230 V-Seite eines 5V-Steckernetzteils manuell ein- und ausschaltbar (Stromersparnis).

Beim morgendlichen Starten gibt es nun 2 Zustände:

a) WLAN ist bereits aktiv, **danach** wird erst die openDTU-Spannung manuell eingeschaltet.
In diesem Fall startet die DTU immer problemlos und alles läuft einwandfrei.

b) Die open DTU wird **zuerest** eingeschaltet, **danach** erst das WLAN. Der Zeitabstand dazwischen kann im Bereich von einigen sec. bis 15 min. liegen. Beim Start in dieser Reihenfolge startet die openDTU **nicht**.
Im Display wird dann angezeigt:
- 0 W
- Heute: 0 Wh
- Ges.: 0.0 kWh
- 01.01.1970  01:01

Die openDTU ist mit Eingabe der IP am Handy über das WLAN nicht ansprechbar (Fehlermeldung), auch nicht bei vielfacher Wiederholung.

Schaltet man die openDTU nun kurz aus und dann wieder ein ist die Verbindung vorhanden und alles läuft einwandfrei.

Hat jemand hier dieses Problem auch oder / und hat ggf. eine Idee, was die Ursache sein könnte?

Danke im Voraus und viele Grüße.

**Comment by tbnobody** [comment]
Wenn kein WLAN mit dem entsprechenden Namen verfügbar ist (dein Fall b) wird nach 15 Sekunden der Verbindungsversuch unterbrochen. Anschließend wird 10 Minuten gewartet und wieder für 15 Sekunden ein Verbindungsversuch unternommen. Also je nach Timing kann es nach dem starten der Fritz!Box schon etwas länger dauern. 

Warum wird das gemacht? Wenn der ESP nach WLANs sucht, scannt er alle möglichen Kanäle durch. Wenn das geschieht ist sein eigener AP auch nur auf diesen Kanälen erreichbar (STA und AP sind immer am gleichen Kanal). Daher könnte man sich ohne das pausieren des Verbindungsvorganges nicht mit dem integrierten AP verbinden.

**Comment by Fokker-1** [comment]
Guten Morgen,

vielen Dank für Deine schnelle und umfangreiche Antwort!

Ich habe das nun mal getestet. Ja, es funktioniert ganau so, wie Du es beschrieben hast!
Man muss nur die notendige Geduld haben ... aber Geduld ist ja immer so ein besonderes Thema ...

Dann ist alles prima, das beobachtete Verhalten entspricht also den Vorgaben.

Schönen Tag noch!
